### DISCUSSION TITLE
Providers not in sync in split screen

### DISCUSSION THREAD

**Discussion by Francan** [rrousselGit/riverpod#2572] [Q&A]
Hello everyone,

I'm new to Flutter and Riverpod, thus I may be doing something terribly wrong.

I've decided to create a button which splits the view into 2. It is implemented by taking the current widget instance, available in the ConsumerState and providing it as argument to a simple widget which then copy the reference into the left and right views, essentially it duplicates the current view. I'm using the instance in an attempt to keep the state in sync between the two views.

There is a core notifier which queries a simulated back-end and retrieves a model class made than available to the main app (naturally a future type). The same notifier allows to update its state through a copyWith method, which overrides the state.

It works, but not consistently, meaning sometimes the changes to the state are correctly reflected on the second view, while other times they do not. Every state change goes through the same copyWith method of the notifier. For example, a button click which change the name of a person, will trigger a callback that through a ref.read of the main provider.notifier calls the copyWith method that updates the state.

I would like to provide some code example, but unfortunately the code cannot be shared. I've already attempted to replicate a similar simpler architecture, yet it seems to work without issue. If there is any snippet that could be helpful I would gladly provided it.

Any help would be greatly appreciated!

**Comment by rrousselGit** [comment]
Without code, there's hardly anything that we can do to help.

It could be due to many things. Like maybe you're missing a ref.watch or you're using the wrong ref or something. Hard to say
